### DISCUSSION TITLE
How to add submenus to context menu entries?

### DISCUSSION THREAD

**Discussion by Rabcor** [lxqt/lxqt#2575] [Q&A] [answered]
I have figured out how to make a context menu in pcmanfm-qt

in `~/.local/share/file-manager/actions/extract.desktop`
```
[Desktop Entry]
Type=Action
Name=Extract Files Here
Icon=archiver
Profiles=profile-zero;

[X-Action-Profile profile-zero]
SelectionCount==1
MimeTypes=application/zip;application/vnd.rar;application/x-compressed-tar;application/x-tarz;application/x-7z-compressed;application/gzip
Exec=~/Scripts/extract.sh %f
Name=Extract
```

And i also found a pretty great example for conditional context menu entries https://forum.slitaz.org/topic/custom-actions-in-the-pcmanfm

But I cannot for the life of me figure out how to make a context menu item that displays a submenu (like say the open with submenu), how do i  do it?

**Comment by tsujan** [comment, marked as answer]
→ https://github.com/lxqt/pcmanfm-qt/wiki/custom_actions#menu-definition

**Reply by Rabcor** [reply]
Thanks! I looked everywhere for that, it's strange that a web search didn't turn it up.
